pkg/apis/trainer/v1alpha1/trainingruntime_types.go

```diff
@@ -44,11 +44,13 @@ const (
 type ClusterTrainingRuntime struct {
 	metav1.TypeMeta `json:",inline"`
 
-	// Standard object's metadata.
+	// metadata of the ClusterTrainingRuntime.
+	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	// Specification of the desired ClusterTrainingRuntime.
-	Spec TrainingRuntimeSpec `json:"spec,omitempty"`
+	// spec of the ClusterTrainingRuntime.
+	// +required
+	Spec TrainingRuntimeSpec `json:"spec,omitzero"`
```

Comment:
Does `omitzero` introduce breaking change ?
I noticed that we disable it for JobSet: https://github.com/kubernetes-sigs/jobset/pull/1046/files

Reply:
I cannot think of any breaking changes. The one case maybe would be when the field has a valid zero value, but that case is also checked now to enforce pointer to differentiate between defaulted zero value and user provided zero value.

Reply:
Why do we need this for `spec` ? I thought this might be useful for types like time.Time.
I am not sure how well it k8s API server works with such fields, did you see other projects also migrate to this new Go 1.24 feature ?

Reply:
The TrainingRuntimeSpec field does not allow the zero value, so the linter mandates to add the tag, and in practice I don't think it changes anything since that non-pointer field can never be zero.

Reply:
it looks like OpenAPI spec generator can't process these values: https://github.com/kubeflow/trainer/blob/cfdfd5989e7a4f383049f0fd0fc8c5605f593fb9/api/python_api/kubeflow_trainer_api/models/trainer_v1alpha1_train_job.py#L35

I should keep `spec` Optional.

Reply:
Good catch, thanks. Updated to keep those fields optional.